node/accounts-bot/index.js

```diff
@@ -0,0 +1,183 @@
+const {google} = require('googleapis');
+const SHEETS_SCOPE = 'https://www.googleapis.com/auth/spreadsheets.readonly';
+const ACCOUNTS_SHEET_ID = '1kxW15ZI48mh4KkvgsMpg7gInmEQmyKYRnZdbUOSMRnU';
+const ACCOUNTS_SHEET_URL = 'https://docs.google.com/spreadsheets/d/1kxW15ZI48mh4KkvgsMpg7gInmEQmyKYRnZdbUOSMRnU/edit#gid=0';
+const ACCOUNT_IMAGE_URL = 'https://www.gstatic.com/images/icons/material/system_gm/1x/account_circle_black_18dp.png';
+
+/**
+ * Looks up the account owner for the company requested.
+ *
+ * @param {!express:Request} req HTTP request context.
+ * @param {!express:Response} res HTTP response context.
+*/
+exports.accountsBot = async (req, res) => {
+  const message = req.query.message || req.body.message;
+  if (!message || !message.text) {
+    console.log('invalid input');
+    res.send(createTextResponse('Please provide an account to look up'));
+  }
+
+  const accountName = getAccountName(JSON.stringify(message.text));
+  console.log(`accountName: ${accountName}`);
+  const sheetsClient = await getSheetsClient();
+  const ownerData = await accountLookup(sheetsClient, accountName);
+  if (ownerData.length > 0) {
+    res.send(createOwnerCard(ownerData));
+  } else {
+    res.send(createErrorCard(accountName));
+  }
+};
+
+/**
+ * Looks up ownership information for the account.
+ *
+ * @param {Object} client initialized Sheets API client.
+ * @param {String} accountName inbound message
+ * @return {String} owner data
+ */
+async function accountLookup(client, accountName) {
+  const accountsSheet = await client.spreadsheets.values.get({
+    spreadsheetId: ACCOUNTS_SHEET_ID,
+    range: 'Sheet1!A:D',
+  });
+  const accountsTable = accountsSheet.data.values;
+  console.log(`accounts: ${JSON.stringify(accountsTable, null, 4)}`);
+
+  let data = [];
+
+  accountsTable.forEach((entry) => {
```

Comment:
Uxe find() instead since you're just looking for one entry:

return accountsTable.find(entry => entry[0] === acountName);

Reply:
good idea! much cleaner